public/app/features/dashboard/components/SaveDashboard/forms/SaveDashboardAsForm.tsx

```diff
@@ -67,15 +65,30 @@ export const SaveDashboardAsForm: React.FC<SaveDashboardFormProps & { isNew?: bo
           },
           dashboard
         );
+
         if (result.status === 'success') {
           onSuccess();
         }
       }}
     >
-      {({ register, control, errors }) => (
+      {({ register, control, errors, getValues }) => (
         <>
-          <Forms.Field label="Dashboard name" invalid={!!errors.title} error="Dashboard name is required">
-            <Input name="title" ref={register({ required: true })} aria-label="Save dashboard title field" autoFocus />
+          <Forms.Field label="Dashboard name" invalid={!!errors.title} error={errors.title?.message}>
+            <Input
+              name="title"
+              ref={register({
+                validate: v => {
+                  if (v.trim() === '') {
+                    return 'Dashboard name is required';
+                  } else if (v && v === getValues().$folder.title?.trim()) {
+                    return 'Dashboard name cannot be the same as folder';
```

Comment:
I never reach this line. This is what I get when logging `getValues` 
![Screenshot from 2020-04-08 10-38-36](https://user-images.githubusercontent.com/1438972/78763248-2f560800-7985-11ea-9745-aff43dd25b22.png)

Reply:
weirdo, will have a look. What exactly did you do?

Reply:
The error appeared when creating a dashboard in the `General` folder with the same name as a top level folder.

Reply:
I did reach this line when I created a dashboard with the same name as a folder. Shoudln't that be possible though?
![Screenshot from 2020-04-08 10-42-03](https://user-images.githubusercontent.com/1438972/78763732-d20e8680-7985-11ea-9f4c-3a997c13bec3.png)

Reply:
this is not possible. See the referenced issue

Reply:
@tskarhed I've tried to reproduce but whad do you mean by "Top level folder"? General is a top level folder. We don't have nested folders structure. So I cannot make the folder title to be undefined.

Reply:
The validation error above will be triggered only when creating a dashboard with the same name as its folder. However, the backend call will also fail if a dashboard is created with the same name as any other existing folder. 
So this catches the error:
![Screenshot 2020-04-09 at 10 20 27](https://user-images.githubusercontent.com/8878045/78868586-dc935380-7a4b-11ea-9621-4269fcc33ef4.png)
While this doesn't:
![Screenshot 2020-04-09 at 10 22 05](https://user-images.githubusercontent.com/8878045/78868667-fa60b880-7a4b-11ea-8fad-36ae9d552c4f.png)

Reply:
@tskarhed this should be fixed now. Do you mind reviewing again?